### PR TITLE
docs: use a processor to transform the operator decision tree data

### PR DESCRIPTION
**Pull request by tmair**
**Description:**
This PR moves the generation of the decision tree into a dgeni processor, so that the decision tree will be generated as part of the documentation generation.

**Related issue (if exists):**
None

### CODE REVIEW

**Comment by benlesh** [comment]
Hey @tmair ! I really appreciate all of this good work. Generally, I'm not the one that reviews anything to do with the docs app, so I've been deferring to @jakovljevic-mladen and @niklas-wortmann for that. This seems like a nice add though.

**Review by niklas-wortmann — APPROVED** [review]
LGTM thanks for adjusting this

**Review by jakovljevic-mladen — CHANGES_REQUESTED** [review]
@tmair, this PR is awesome, thank you a lot for merging Operator Decision Tree generator to the tools that already exist.

I have one comment regarding a file rename; everything else looks great.

File: apps/rxjs.dev/tools/transforms/config.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ const DOCS_OUTPUT_PATH = resolve(OUTPUT_PATH, 'docs');
 const API_SOURCE_PATH = resolve(PROJECT_ROOT, 'packages/rxjs/src');
 const MARBLE_IMAGES_PATH = resolve(SRC_PATH, 'assets/images/marble-diagrams');
 const MARBLE_IMAGES_WEB_PATH = 'assets/images/marble-diagrams';
+const DECISION_TREE_PATH = resolve(CONTENTS_PATH, 'tree.yml');
```
[the remark this answers is not in the document]
Comment (tmair) [reply apps/rxjs.dev/tools/transforms/config.js]: Yes, that is a good remark. I renamed the file.

**Comment by tmair** [comment]
> I have one comment regarding a file rename; everything else looks great. Also, conflicts need to be resolved.

I rebased the PR and addressed the comment about the renaming of the file.

**Review by jakovljevic-mladen — APPROVED** [review]
LGTM. Thanks a lot for making these changes @tmair 🎉

**Pull request merged**